web_widget_text_markdown/static/src/js/web_widget_text_markdown.js

```diff
@@ -0,0 +1,116 @@
+    'use strict';
+
+    var basic_fields = require('web.basic_fields');
+    var field_registry = require('web.field_registry');
+    var core = require('web.core');
+
+    var _t = core._t;
+    var LIBS_PATH = '/web_widget_text_markdown/static/src/lib';
+
+
+    var FieldTextMarkDown = basic_fields.FieldText.extend({
+        jsLibs: [
+            LIBS_PATH + '/marked.js',
+            LIBS_PATH + '/dropzone.js',
+            LIBS_PATH + '/bootstrap-markdown.js',
+        ],
+        cssLibs: [
+            LIBS_PATH + '/bootstrap-markdown.min.css',
+        ],
+
+        _getValue: function () {
+            return this.$markdown.getContent();
+        },
+
+        _prepareInput: function () {
+            var $input = this._super.apply(this, arguments);
+            _.defer(function ($elm) {
+                $elm.markdown(this._getMarkdownOptions());
+                this.$markdown = $elm.data("markdown");
+                this.$markdown.setContent(this.value || "");
+            }.bind(this), $input);
+            return $input;
+        },
+
+        _renderReadonly: function () {
+            this.$el.html(marked(this._formatValue(this.value)));
+        },
+
+
+        _getMarkdownOptions: function () {
+            var markdownOpts = {
+                autofocus: false,
+                savable: false,
+            };
+
+            // Only can create attachments on created records
+            if (this.res_id) {
+                markdownOpts.dropZoneOptions = {
+                    paramName: 'ufile',
+                    url: '/web_widget_text_markdown/upload_attachment',
+                    acceptedFiles: 'image/*',
+                    params: {
+                        csrf_token: core.csrf_token,
+                        session_id: this.getSession().override_session,
+                        model: this.model,
+                        res_id: this.res_id,
+                    },
+                    success: this._markdownUploadSuccess.bind(this),
+                    error: this._markdownUploadError.bind(this),
+                };
+
+                if (_t.database.multi_lang && this.field.translate) {
+                    markdownOpts.additionalButtons = [
+                        [{
+                            name: 'oTranslate',
+                            data: [{
+                                name: 'cmdTranslate',
+                                title: _t('Translate'),
+                                icon: {glyph: 'glyphicon glyphicon-flag'},
+                                callback: this._markdownTranslate,
+                            }],
+                        }],
+                    ];
+                }
+            }
+
+            return markdownOpts;
+        },
+
+        _markdownUploadSuccess: function () {
+            this.isDirty = true;
+            this._doDebouncedAction();
+            $(".dz-success-mark svg").css({
```

Comment:
can we have SCSS file for this?
This way you can use standard odoo colors (or theme colors) and ease customization.
Not mandatory.

Reply:
Ok! but v11 uses only less no?